src/app/account/keys/layout.tsx

```diff
@@ -0,0 +1,26 @@
+import { getReviewerPublicKeyAction } from '@/server/actions/user-keys.actions'
+import { UserLayout } from '@/components/layout/user-layout'
+import { AnonLayout } from '@/components/layout/anon-layout'
+import { ReactNode } from 'react'
+import { RegenerateKeys } from './regenerate-keys'
+
+export default async function ReviewerKeysPageLayout({
+    children,
+}: Readonly<{
+    children: ReactNode
+}>) {
+    const publicKey = await getReviewerPublicKeyAction()
+    const clerkPublishableKey = process.env.NEXT_PUBLIC_CLERK_PUBLISHABLE_KEY || ''
+
+    if (publicKey) {
+        // User has a key - show UserLayout
+        return (
+            <UserLayout>
+                <RegenerateKeys />
+            </UserLayout>
+        )
+    } else {
+        // No key, first-time generation - show AnonLayout
+        return <AnonLayout clerkPublishableKey={clerkPublishableKey}>{children}</AnonLayout>
+    }
+}
```

Comment:
future work: I think we're also reading the `process.env.NEXT_PUBLIC_CLERK_PUBLISHABLE_KEY` in other layouts, lets split AnonLayout so part can be 'use server' and read the env var itself, that way we're not repeating this everywhere